src/ibc_api/metadata.py

```diff
@@ -124,17 +122,25 @@ def fetch_remote_file(
     str
         full path of the fetched file
     """
-    # Link to the json file on the IBC docs repo
-    remote_file = f"{remote_root}/{file}"
-    # save the file locally
-    save_as = os.path.join(local_root, file)
-    # use curl with github api to download the file
-    os.system(
-        f"curl -s -S -L -H 'Accept: application/vnd.github.v4.raw' -H 'X-GitHub-Api-Version: 2022-11-28' {remote_file} -o '{save_as}'"
-    )
-
-    # Return the data
-    return save_as
+    # Construct the url
+    url = f"{remote_root}/{file}"
+
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+
+        # Save the file locally
+        local_file = os.path.join(local_root, file)
+        with open(local_file, "wb") as f:
+            for chunk in r.iter_content(chunk_size=512):
+                if chunk:
+                    f.write(chunk)
+
+        return local_file
+
+    except requests.exceptions.HTTPError as err:
+        print(f"Error fetching {file}: {err}")
+        return None
```

Comment:
```suggestion
        raise(f"Error fetching {file}: {err}")
```
This would directly raise the error here instead of causing further unrelated errors downstream.